selftests/unit/test_utils_partition.py

```diff
@@ -45,12 +37,8 @@ class TestPartition(unittest.TestCase):
 
     @unittest.skipIf(missing_binary('mkfs.ext2'),
                      "mkfs.ext2 is required for these tests to run.")
-    @unittest.skipIf(missing_binary('sudo'),
+    @unittest.skipIf(not process.can_sudo(),
```

Comment:
Btw the issue I have is when avocado fails to find `sudo` binary and not when the `sudo` is configured incorrectly. So let's combine the two approaches and get the best of both of them, OK? I'll adjust all tests as soon as I get your ack on this new approach.